### PR TITLE
Add build_model function for minimal autoencoder

### PR DESCRIPTION
**Pull request by AnnaKwa**
The reservoir training now relies on the autoencoder to do the feature concatentation and normalization/denormalization, which makes the autoencoder a nonoptional component in the reservoir training. This PR adds a function to the autoencoder module to build a minimal autoencoder that has no trainable parameters and only handles the necessary concatenation and normalization. It can be used in the case of no autoencoder path being specified, ex. in tests.
  

- [x] Tests added
 

Coverage reports (updated automatically):

### CODE REVIEW

File: external/fv3fit/fv3fit/reservoir/autoencoder.py [hunk 1]
```diff
@@ -181,6 +181,44 @@ def init_testing(
...
+        )
+        start_ind += output.shape[-1]
+    denorm_output_layers = standard_denormalize(
+        names=variables, layers=decoder_outputs, arrays=X,
+    )
+    decoder = tf.keras.Model(inputs=decoder_input, outputs=denorm_output_layers)
+
+    model = Autoencoder(encoder=encoder, decoder=decoder)
+    # need to call model once so it can save without compiling
+    model([np.ones((3, arr.shape[-1])) for arr in X])
```
Comment (spencerkclark) [inline external/fv3fit/fv3fit/reservoir/autoencoder.py:218]: As discussed on Slack, I like your idea of using `model.compile()` here a little better.  Could consider writing a test for savability if this is an important property of what this function returns as well.

**Review by spencerkclark — APPROVED** [review]
Looks good to me.  Hopefully this makes writing tests simpler going forward!
